Analysis/compare_dynasim_cps.qmd

```diff
@@ -0,0 +1,197 @@
+  'PERNUM', 
+  'DOBY', 
+  'SEX', 
+  'ETHNCTY', 
+  paste0('GRADECAT', 2006:last_year),
+  paste0('NONHOUSW', 2006:last_year),
+  paste0('EARNINGS', 2006:last_year),
+  paste0('SPOUSERN', 2006:last_year),
+  paste0('WEDSTATE', 1951:last_year)
+  )
+
+```
+
+```{r}
+fehdir = "S:/damir/run_1004/"
+fehfile <- paste0(fehdir, "dynasipp_person_even.dat")
+codebookfile = paste0(fehdir, "codebook_2100ds.sipp2006")
+dyndf = read_feh(codebookfile, fehfile, columns=keep_cols) 
+
+# Wide to long format
+df = feh_wide_to_long(dyndf)
+write_rds(df, '../data/dynasim_output_long.Rds')
+```
+
+###Read in CPS ASEC
+```{r}
+asec_df_long2 <- read_rds('../data/asec_longitudinal.Rds')
+
+```
+
+### Cleaning for Viz
+```{r}
+# Creating comparable educational categories, race/ethnicity (possibly age too)
+dyn_long <- read_rds('../data/dynasim_output_long.Rds')|>
+    mutate(age=year-doby,
+           asecwt = 1)|>
+    filter(!is.na(gradecat), age > 14)|> # note: 50% of gradecat missing
+     mutate(educ_clean = as.factor(
+               case_when(
+                  gradecat %in% 0:11 ~ "0-11th grades",
+                  gradecat == 12 ~  "HS degree", 
+                  gradecat == 13 ~ "Some college/No Degree",
+                  gradecat %in% c(14,15) ~ "Associate's Degree",
+                  gradecat == 16 ~ "Bachelor's Degree",
+                  TRUE ~ "More than Bachelor's"
+               )), 
+           race_eth = as.factor(
+               case_when(
+                ethncty %in% c(4000,4999) ~ "Asian",
+                ethncty %in% c(2000, 2999) ~ "Black",
+                ethncty %in% c(1014, 1020, 2014, 2020, 3014, 3020, 4014, 4020) ~ "Hispanic",   
+                ethncty %in% c(1000, 1999) ~ "White",
+                  TRUE ~ "Other"
+               )),
+           age_cat = as.factor(
+               case_when(
+                  age < 25 ~ "Under 25",
+                  age < 36 ~ "26-35",
+                  age < 46 ~ "36-45",
+                  age < 56 ~ "46-55",
+                  age < 66 ~ "56-65",
+                   TRUE ~ "Over 65"
+               )),
+            sex = as.factor(
+               case_when(
+                  sex == 1 ~ "Male",
+                  TRUE ~ "Female"
+               )))
+
+dyn_long$educ_clean <- factor(dyn_long$educ_clean, levels = c("0-11th grades", "HS degree", "Some college/No Degree", "Associate's Degree", "Bachelor's Degree", "More than Bachelor's"))
+
+dyn_long$age_cat <- factor(dyn_long$age_cat, levels = c("Under 25", "26-35", "36-45", "46-55", "56-65", "Over 65"))
+
+# Missings in CPS ASEC - about 24% excluded
+length(asec_df_long2$EDUC[asec_df_long2$EDUC==1])/nrow(asec_df_long2)
+length(asec_df_long2$EDUC[asec_df_long2$EDUC==999])/nrow(asec_df_long2)
+
+asec_df_long <- asec_df_long2 |>
+    filter(EDUC != 999 & EDUC != 1)|> #excluding missings, blanks or NIUs
+    mutate(
+        educ_clean = as.factor(
+               case_when(
+                  EDUC >= 0 & EDUC < 70 ~ "0-11th grades",
```

Comment:
We should take a look at the education encoding because the some college and associate degree categories have discontinuities in the 90s when CPS changed categories.

Reply:
Thanks Damir - here's my understanding from our last call:

Reply:
Thanks Damir - here's my understanding from our last call (and from the [IPUMS documentation](https://cps.ipums.org/cps-action/variables/EDUC#description_section)):

- EDUC is a combination of two variables. The codes for this variable include some college is 080 and 081 (though only 081 seems to have any data entries); and that associate degree categories are 090, 091, 092, 100 (though only 091 and 092 seems to have any data entries)
- The first variable which is used to create EDUC, HIGRADE, is available prior to 1992. It is not immediately clear to me how some college or associates degrees were coded, but I had assumed that it was the same as the coding reflected in EDUC. From IPUMS: "Prior to 1992, while it is known how many years of higher education was completed, there is no way to know definitively whether completing these years resulted in a degree." Seems like we are basically assuming that if someone had 2 or more years of college but no bachelor's, we will be assuming that they have an associate's degree (which may not be true prior to 1992).
- The second variable, [EDUC99](https://cps.ipums.org/cps-action/variables/EDUC99#codes_section), is available after 1992. There seems to be one category for some college, no degree, and three for associate's degrees (Associate degree, type of program not specified; Associate degree, occupational program; Associate degree, academic program). These seem to match the 091 and 092 codes in EDUC.